pkgs/by-name/to/todesk/package.nix

```diff
@@ -0,0 +1,173 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  dpkg,
+  autoPatchelfHook,
+  nspr,
+  kmod,
+  systemdMinimal,
+  glib,
+  bubblewrap,
+  libX11,
+  libXrandr,
+  glibc,
+  libdrm,
+  coreutils,
+  libGL,
+  bash,
+  libXcomposite,
+  libXdamage,
+  libXfixes,
+  libXtst,
+  nss,
+  libXxf86vm,
+  gtk3,
+  gdk-pixbuf,
+  pango,
+  appindicator-sharp,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "todesk";
+  version = "4.7.2.0";
+
+  src = fetchurl {
+    url = "https://newdl.todesk.com/linux/todesk-v${finalAttrs.version}-amd64.deb";
+    hash = "sha256-v7VpXXFVaKI99RpzUWfAc6eE7NHGJeFrNeUTbVuX+yg=";
+    curlOptsList = [
+      "--user-agent"
+      "Mozilla/5.0"
+    ];
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+  ];
+  buildInputs = [
+    nspr
+    kmod
+    systemdMinimal
+    glib
+    bash
+    coreutils
+    libX11
+    libXrandr
+    glibc
+    libdrm
+    libGL
+    libXcomposite
+    libXdamage
+    libXfixes
+    libXtst
+    nss
+    libXxf86vm
+    gtk3
+    gdk-pixbuf
+    pango
+    appindicator-sharp
+  ];
+
+  autoPatchelfIgnoreMissingDeps = [
+    "iHD_drv_video.so"
+    "libglut.so"
+    "libigdgmm.so"
+    "libmfx.so"
+    "libmfxhw64.so"
+    "libva.so"
+    "libva-drm.so"
+    "libva-x11.so"
+    "libzrtc.so"
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    dpkg -x $src ./todesk-src
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir "$out"
+    cp -r todesk-src/* "$out"
+    mv "$out/opt/todesk/bin" "$out/bin"
+    # todesk puts their config into /opt/todesk, however we haven't got this. So we just use bubblewrap to create a correct /opt/todesk for todesk
+    echo "#! ${bash}/bin/bash -e" > "$out/bin/ToDesk-Wrap"
+    echo "if [ ! -d /var/lib/todesk ]; then">> "$out/bin/ToDesk-Wrap"
+    echo "mkdir /var/lib/todesk" >> "$out/bin/ToDesk-Wrap"
+    echo "fi">> "$out/bin/ToDesk-Wrap"
```

Comment:
Please use `pkgs.writeShellScript`.

Reply:
You could use `let .. in` to write a shell script in the `let` instruction to clean up these `echo` statements. However `writeShellScript` works just as well to remove all those `echo` statements